File: android/app/src/main/java/com/itcorner/moodleandroid/data/dto/CourseDto.kt
package com.itcorner.moodleandroid.data.dto

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class CourseDto(
    val id: Int,
    val fullname: String? = null,
    val displayname: String? = null,
    val progress: Int? = null,
    val completionprogress: Int? = null,
    val courseimage: String? = null,
    val overviewfiles: List<OverviewFileDto>? = null
)

@JsonClass(generateAdapter = true)
data class OverviewFileDto(
    val fileurl: String? = null
)

File: ios/MoodleAssessmentApp/MoodleAssessmentApp/Data/DTO/CourseDTO.swift
import Foundation

struct CourseDTO: Decodable {
    let id: Int
    let fullname: String?
    let displayname: String?
    let progress: Int?
    let completionprogress: Int?
    let courseimage: String?
    let overviewfiles: [OverviewFileDTO]?

    init(
        id: Int,
        fullname: String?,
        displayname: String?,
        progress: Int?,
        completionprogress: Int?,
        courseimage: String? = nil,
        overviewfiles: [OverviewFileDTO]?
    ) {
        self.id = id
        self.fullname = fullname
        self.displayname = displayname
        self.progress = progress
        self.completionprogress = completionprogress
        self.courseimage = courseimage
        self.overviewfiles = overviewfiles
    }
}

struct OverviewFileDTO: Decodable, Equatable {
    let fileurl: String
}
